### PR TITLE
TPL build instructions

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR involves documentation of the TPL build/update process in the Axom Dev Guide.
- It clarifies some points and corrects some script names, directory names, etc.

The relevant section of generated docs for this PR is here: https://axom.readthedocs.io/en/task-rhornung67-fix-devdocs/docs/sphinx/dev_guide/updating_tpls.html

### CODE REVIEW

**Review by gunney1 — APPROVED** [review]
Thank you @rhornung67

File: src/docs/sphinx/dev_guide/updating_tpls.rst [hunk 1]
```diff
@@ -189,24 +194,24 @@ More info on :ref:`building-axom-label`.
 Built-in TPLs
 -------------
 
-Axom several lightweight header-only libraries that we use internally and
-expose for downstream customers to use if they wish.
+Axom uses several, lightweight, header-only libraries internally which are
+exposed for downstream customers to use if they wish.
```
Comment (white238) [inline src/docs/sphinx/dev_guide/updating_tpls.rst:198]:
```suggestion
Axom uses several lightweight, header-only libraries internally, which are
exposed for downstream customers to use if they wish.
```

I think this is more correct. maybe... the comma after several weirds me out but Word doesn't complain about it.

**Review by white238 — APPROVED** [review]
Thanks @rhornung67 !

**Review by kennyweiss — APPROVED** [review]
Thanks @rhornung67!

File: src/docs/sphinx/dev_guide/updating_tpls.rst [hunk 2]
```diff
@@ -134,24 +134,29 @@ the following:
...
+are designed to handle building suites of TPLs via Uberenv and Spack.
+
+* ``build_tpls.py``: This script starts by building all TPLs listed in the file
+  ``specs.json``. It will generate host-config files and copy them to the base
+  of the Axom repository. After building all of the TPLs, it will test Axom
+  against those built TPLs as well as test the installed ``using-with-cmake``
+  example for correctness. This script stops at the first failed TPL build but
+  attempts to build all host-configs against the Axom source with a summary at
+  the end of which succeeded or failed.
+* ``build_src.py``: This scripts takes the existing host-configs, or the
```
Comment (agcapps) [inline src/docs/sphinx/dev_guide/updating_tpls.rst:152]: Typo: "This scripts" should be "this script"

**Review by agcapps — APPROVED** [review]
Thank you, Rich.

File: src/docs/sphinx/dev_guide/updating_tpls.rst (hunk 2 above)
Comment (rhornung67) [reply src/docs/sphinx/dev_guide/updating_tpls.rst:152]: Thank you. Fixed pushed.

**Pull request merged**